filament/backend/src/vulkan/VulkanContext.h

```diff
@@ -37,15 +39,77 @@ struct VulkanTexture;
 class VulkanStagePool;
 struct VulkanTimerQuery;
 
+// TODO: We want to use std::shared_ptr across the driver, but VulkanTexture also maps to
+// HwTexture so it's not possible. We can clean this up (remove the variant) once we break that
+// mapping.
+struct TexturePointer {
+    TexturePointer() = default;
+    explicit TexturePointer(VulkanTexture* tex) :mTexture(tex) {}
+    explicit TexturePointer(std::shared_ptr<VulkanTexture> tex) :mTexture(tex) {}
+
+    inline void operator=(VulkanTexture* tex) {
```

Comment:
nit: return `TexturePointer&` for `operator=` methods?

Reply:
done